src/test/java/org/terasology/launcher/repositories/JenkinsClientTest.java

```diff
@@ -46,8 +48,10 @@ void canHandleInvalidJsonPayload() throws IOException {
         InputStream invalidPayload = new ByteArrayInputStream("{ this is ] no json |[!".getBytes());
 
         URL urlToInvalidPayload = mock(URL.class);
-        doReturn(invalidPayload).when(urlToInvalidPayload).openStream();
 
-        assertNull(client.request(urlToInvalidPayload));
+        try (MockedStatic<JenkinsClient> utilities = Mockito.mockStatic(JenkinsClient.class)) {
+            utilities.when(() -> JenkinsClient.openStream(urlToInvalidPayload)).thenReturn(invalidPayload);
```

Comment:
This is a little different than the other mockito usages I've seen. Is it because it's a static function, and mocking those works differently?

Reply:
Yep, exactly. I was looking at https://www.baeldung.com/mockito-mock-static-methods for reference. 

I feel uneasy to mock (parts of) the unit under test, and using a static method that I can stub separately looked like a good compromise to keep the test.